### DISCUSSION TITLE
Default Location of Caddy TLS / SSL Certificate, and Private Keys

### DISCUSSION THREAD

**Discussion by yaq-cc** [coder/code-server#6458] [General]
Hi all,

I'd like to reuse the existing Certificate and Key to host a TLS / SSL server for testing purposes.  Caddy is hosting my code-server instance - what are my options?

**Comment by code-asher** [comment]
If you have a domain I would use Let's Encrypt which Caddy automatically supports.

code-server generates certificates in the XDG data directory I believe, `~/.local/share/code-server` by default.  But if you have no domain I would probably just set `tls internal` in the Caddy config and let it generate its own self-signed certificates rather than use the ones code-server generates.

**Reply by code-asher** [reply]
And if you are asking the location of Caddy's certificates rather than code-server's certificates, I have no idea but you can probably find that information in Caddy's docs or by asking the Caddy folks.

**Comment by kaze251** [comment]
><
